### PR TITLE
Players cannot observe before PREGAME is reached, Version 2

### LINKED ISSUE #28157: People who observe before the mapping(?) subsystem kicks in aren't initialized
kind of an issue
@Cyberboss

### LINKED ISSUE #28205: If you observe fast enough, lobby music will play in game
This is perturbing

### PR DESCRIPTION
**Pull request by optimumtact**
Reverts tgstation/tgstation#28229

Fixes #28157
Fixes #28205

### CODE REVIEW

File: code/modules/mob/dead/new_player/new_player.dm [hunk 1]
```diff
@@ -29,21 +29,25 @@
 /mob/dead/new_player/prepare_huds()
 	return
 
-/mob/dead/new_player/proc/new_player_panel()
+/mob/dead/new_player/proc/linkify_ready(string, value)
+	return "<a href='byond://?src=\ref[src]:ready=[value]>[string]</a>"
 
-	var/output = "<center><p><a href='byond://?src=\ref[src];show_preferences=1'>Setup Character</A></p>"
+/mob/dead/new_player/proc/new_player_panel()
+	var/output = "<center><p>< href='byond://?src=\ref[src];show_preferences=1'>Setup Character</A></p>"
```
Comment (lzimann) [inline code/modules/mob/dead/new_player/new_player.dm]: missing the `a`

**Comment by optimumtact** [comment]
I'm about to go on holiday til monday, so will finish then.

**Comment by optimumtact** [comment]
Tested and ready now, note it still asks you if you want to observe and bounces you back to not ready if you choose you're not.

new player panel with new options

late join panel is untouched

File: code/modules/mob/dead/new_player/new_player.dm [hunk 2]
```diff
@@ -29,21 +29,24 @@
...
+			if(PLAYER_READY_TO_OBSERVE)
+				output += "<p>\[ [linkify_ready("Ready", PLAYER_READY_TO_PLAY)] | [linkify_ready("Not Ready", PLAYER_NOT_READY)] | <b> Observe </b> \]</p>"
 	else
-		output += "<p><a href='byond://?src=\ref[src];manifest=1'>View the Crew Manifest</A></p>"
-		output += "<p><a href='byond://?src=\ref[src];late_join=1'>Join Game!</A></p>"
-
-	output += "<p><a href='byond://?src=\ref[src];observe=1'>Observe</A></p>"
+		output += "<p><a href='byond://?src=\ref[src];manifest=1'>View the Crew Manifest</a></p>"
+		output += "<p><a href='byond://?src=\ref[src];late_join=1'>Join Game!</a></p>"
+		output += "<p>[linkify_ready("Observe", 2)]</p>"
```
Comment (AnturK) [inline code/modules/mob/dead/new_player/new_player.dm]: Define in here ?

File: code/modules/mob/dead/new_player/new_player.dm [hunk 3]
```diff
@@ -29,21 +29,24 @@
...
-
-	var/output = "<center><p><a href='byond://?src=\ref[src];show_preferences=1'>Setup Character</A></p>"
-
-	if(!SSticker || SSticker.current_state <= GAME_STATE_PREGAME)
-		if(ready)
-			output += "<p>\[ <b>Ready</b> | <a href='byond://?src=\ref[src];ready=0'>Not Ready</a> \]</p>"
-		else
-			output += "<p>\[ <a href='byond://?src=\ref[src];ready=1'>Ready</a> | <b>Not Ready</b> \]</p>"
+/mob/dead/new_player/proc/linkify_ready(string, value)
+	return "<a href='byond://?src=\ref[src];ready=[value]'>[string]</a>"
```
Comment (lzimann) [inline code/modules/mob/dead/new_player/new_player.dm]: the linkify could be a macro

**Comment by optimumtact** [comment]
DNM, now I have to test again

**Comment by optimumtact** [comment]
Tested and working post macro changes

**Comment by Bawhoppen** [comment]
this is horrible why did we do this

**Pull request merged**